### ISSUE TITLE
req.getRequestDispatcher("/test.htm").include(req, resp); does not go anywhere on jetty12ee10 on exploded deploy, works on ee9 and older jettys.

### ISSUE THREAD

**Issue by andresluuk** [eclipse/jetty.project#10142]
**Jetty version(s)**
jetty12 built from source
**Java version/vendor** `(use: java -version)`
JDK 17
**OS type/version**
Windows/Unix
**Description**
Maybe the same as https://github.com/eclipse/jetty.project/issues/10139 ?
**How to reproduce?**
I have a simple servlet that does forwarding into a local htm file:

``` java
	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException {
		System.out.println("MyServlet.doGet() called");
		try {
		  req.getRequestDispatcher("/test.htm").include(req, resp);
		} catch (IOException e) {
			throw new ServletException(e);
		}
	}
```

test.htm is in the root of the web app.
If I deploy the application as packaged application on jetty12ee10 then it works and test.htm is shown. But If I deploy it as exploded application then I will get an empty page. The same application for jett12ee9 it works in both modes.

**Comment by joakime** [comment]
How is this servlet mapped?
Can you detail your `<servlet>` and `<servlet-mapping>` entries for the above Servlet?
And then what your request is, along with any context-path information for the containing `WebAppContext` or `ServletContextHandler`?

**Comment by andresluuk** [comment]
The request url is: http://localhost:8080/exp-servletContext-requestDispatcherInclude/myservlet/xxx?3731317427365001949

web.xml is:
```
<web-app xmlns="http://java.sun.com/xml/ns/j2ee"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://java.sun.com/xml/ns/j2ee 
	      http://java.sun.com/xml/ns/j2ee/web-app_2_4.xsd"
	version="2.4">

  <servlet>
    <servlet-name>myServlet</servlet-name>
    <servlet-class>test.MyServlet</servlet-class>
  </servlet>
  
  <servlet-mapping>
    <servlet-name>myServlet</servlet-name>
    <url-pattern>/myservlet/*</url-pattern>
  </servlet-mapping>

  <welcome-file-list>
    <welcome-file>/myservlet</welcome-file>
  </welcome-file-list>
</web-app>
```
I can send the entire test application if needed, but there is not much more there.

**Comment by andresluuk** [comment]
One more note here. It actually fails with ee10 in both modes on plain jetty run, but passes with ee9 in both modes. (It passed on my test before because of jrebel).

http://localhost:8080/pkg-servletContext-requestDispatcherInclude/myservlet/xxx?59122691717

[pkg-servletContext-requestDispatcherInclude.zip](https://github.com/eclipse/jetty.project/files/12164462/pkg-servletContext-requestDispatcherInclude.zip)

**Comment by joakime** [comment]
I have a testcase in PR #10149 that uses your setup (in comment https://github.com/eclipse/jetty.project/issues/10142#issuecomment-1650308023) and passes (but needs the fixes in that PR to work)

**Comment by andresluuk** [comment]
Sad news here. I used they jetty built from the PR#10149 but this case still fails.
One difference between the 2 cases is, that this one does not have the context-param. (I added it for testing now, but it did not change anything). I think the welcome-file part is not relevant to this case so the nullpointer fix was for this case?

**Comment by gregw** [comment]
I've reproduced the problem in branch: https://github.com/eclipse/jetty.project/tree/fix/12.0.x/10142-include-static, based off the PR #10149 (but I don't think it is related).
@lorban do you want to look at fixing it, as I think it is probably related to the simplification I suggested to you

**Comment by joakime** [comment]
PR #10154 opened for this issue

**Comment by lorban** [comment]
@andresluuk Would it be possible for you to test that the changes in PR #10140 do solve this problem?

**Comment by andresluuk** [comment]
I think the PR got merged? So I tryed out 12.0.x and my test passed on there. Thanks!

**Comment by lorban** [comment]
@andresluuk Yes, https://github.com/eclipse/jetty.project/pull/10140 got merged shortly after I sent you that message.

I'm happy that this change made your test pass, thanks for your feedback!
